File: odd-even-linked-list/odd-even-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) {
        if(head == null || head.next == null)
        {
            return head;
        }
        ListNode odd = head;
        ListNode even = head.next;
        ListNode remember = even;
        while(even!=null && even.next!=null)
        {
            ListNode temp = even.next;
            odd.next = temp;
            even.next = temp.next;
            if(odd.next!=null)
            {
                odd = odd.next;
            }
             if(even.next!=null)
            {
                even = even.next;
            }
        }
        odd.next = remember;
        return head;
    }
}

File: odd-even-linked-list/odd-even-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head or not head.next:
            return head;
        
        odd = head;
        stored = head.next;
        even = head.next;
        
        while odd and even and even.next:
            oddNxt = odd.next.next;
            evenNxt = even.next.next;
            odd.next = oddNxt;
            even.next = evenNxt;
            odd = odd.next;
            even = even.next;
        odd.next = stored;
        return head;
            
            
        